modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordApi.kt

```diff
@@ -87,58 +123,61 @@ class MirrordApi(private val service: MirrordProjectService) {
         configFile: String?,
         wslDistribution: WSLDistribution?
     ): List<String> {
-        logger.debug("listing pods")
-
-        val commandLine = GeneralCommandLine(cli, "ls", "-o", "json")
-        configFile?.let {
-            logger.debug("adding configFile to command line")
-            commandLine.addParameter("-f")
-            val formattedPath = wslDistribution?.getWslPath(it) ?: it
-            commandLine.addParameter(formattedPath)
+        val task = MirrordLsTask(cli).apply {
+            this.configFile = configFile
+            this.wslDistribution = wslDistribution
+            this.output = "json"
         }
 
-        wslDistribution?.let {
-            logger.debug("patching to use WSL")
-            val wslOptions = WSLCommandLineOptions()
-            wslOptions.isLaunchWithWslExe = true
-            it.patchCommandLine(commandLine, service.project, wslOptions)
-        }
-
-        logger.debug("creating command line and executing $commandLine")
-
-        val process = commandLine.toProcessBuilder()
-            .redirectOutput(ProcessBuilder.Redirect.PIPE)
-            .redirectError(ProcessBuilder.Redirect.PIPE)
-            .start()
-
-        logger.debug("waiting for process to finish")
-        process.waitFor(60, TimeUnit.SECONDS)
-
-        logger.debug("process wait finished, reading output")
+        return task.run(service.project)
+    }
 
-        if (process.exitValue() != 0) {
-            val processStdError = process.errorStream.bufferedReader().readText()
-            throw MirrordError.fromStdErr(processStdError)
-        }
+    private class MirrordExtTask(cli: String) : MirrordCliTask<MirrordExecution>(cli, "ext") {
+        override fun compute(project: Project, process: Process, setText: (String) -> Unit): MirrordExecution {
+            val parser = SafeParser()
+            val bufferedReader = process.inputStream.reader().buffered()
+
+            val warningHandler = MirrordWarningHandler(project.service<MirrordProjectService>())
+
+            setText("mirrord is starting...")
+            for (line in bufferedReader.lines()) {
+                Thread.sleep(1000)
```

Comment:
Why the sleep?

Reply:
Thanks! Leftover, had trouble with cancelling on time